nbinteract/questions.py

```diff
@@ -65,53 +81,72 @@ def check_answer(answer_content):
+def short_answer(question, answers, explanation=None):
     """
     Generates a short answer question that allows user to input an answer in
     a textbox and a submit button to check the answer.
 
     Args:
         question (str):  The question being asked.
 
-        answer_list (str):  A list of correct answers.
-
-        check_function (func): A predicate function that returns True for
-            correct answers.
+        answers (str | list str | func): If a string, only that string will be
+            marked correct. If a list of string, any string in the list will be
+            marked correct. If a function, any input that causes the function
+            to return True will be marked correct.
 
         explanation (str):  The explanation to the question is displayed when
             the user inputs the correct answer.
 
     Returns:
         None
 
-    >>> short_answer('What is 1+1?', '2') #doctest: +SKIP
+    >>> short_answer('What is 1 + 1?', '2',
+    ...              explanation='1+1 is 2') #doctest: +SKIP
     <What is 1+1?>
     <Input box, Submit button>
+    >>> short_answer('Enter the first name of a member of the Beatles.',
+    ...              ['John', 'Paul', 'George', 'Ringo']) #doctest: +SKIP
+    <Enter the first name of a member of the Beatles.>
+    <Input box, Submit button>
+    >>> short_answer('Enter an even number.',
+    ...              lambda x: int(x) % 2 == 0) #doctest: +SKIP
+    <Enter an even number.>
+    <Input box, Submit button>
     """
     # Input textbox
-    user_answer = widgets.Text(placeholder='Write your answer here')
+    textbox = widgets.Text(placeholder='Write your answer here')
     # Submit button
     submit_button = widgets.Button(description='Submit')
     # Space right of the submit button to show checkmark/x-mark
     visual_correct = widgets.HTML()
+    # Space below input line to display error if function call errored
+    error_space = widgets.HTML()
     # Space below input line to display explanation if answer is correct
-    explanation_space = widgets.HTML()
+    explain_space = widgets.HTML()
 
     # correctness function linked to the submit button
-    def check_answer(text_content):
-        try:
-            user_answer_value = int(user_answer.value)
-        except ValueError:
-            user_answer_value = user_answer.value
-        if user_answer_value in answer_list or check_function(user_answer_value):
-            visual_correct.value = CHECK_ICON
-            if explanation:
-                explanation_space.value = explanation
+    def check_answer(_):
+        response = textbox.value
+        if isinstance(answers, collections.Callable):
+            try:
+                error_space.value = ''
+                correct = answers(response)
+            except Exception as e:
+                correct = False
+                error_space.value = 'Error in checking answer: {}'.format(e)
+        elif isinstance(answers, str):
```

Comment:
do we expect them to give the answer as a str even if it's a number then? Since the textbox input is always string if you don't convert the choice to str (or the input to int) it'll be incorrect I think.

Reply:
Yeah I saw your code to try casting to an int. I think the issue with that is that you can get some unexpected results when you use functions to check the answer since sometimes the input will be a number and sometimes the input will be a string depending on what the user inputs. I think it's easier for users to just know that the user input is always a string and act accordingly.